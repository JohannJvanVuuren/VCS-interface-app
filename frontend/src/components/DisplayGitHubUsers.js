/* Import of the main stylesheet generated by the SCSS files and preprocessor */
import '../sass/main.css';

/* Import of the useLocation hook which will enable to get the state passed via the Link components in
* the Header component */
import {useLocation} from 'react-router-dom';
import {useEffect, useState} from "react";
import ReactPaginate from 'react-paginate';


/* Definition of the DisplayGitHubUsers component */
export const DisplayGitHubUsers = () => {

    /* Declaration and initialisation of state variables */
    const [users, setUsers] = useState([]);
    const [isLoading, setIsLoading] = useState(false);
    const [currentUsers, setCurrentUsers] = useState(null);
    const [pageCount, setPageCount] = useState(0);
    const [usersOffset, setUsersOffset] = useState(0);

    /* Setting the user that will be displayed per page. This can be changed at a later stage so that the
    * value is obtained from the user */
    const usersPerPage = 10;

    /* Creation of an instance of useLocation */
    const location = useLocation();
    /* Storing the state data from the useLocation instance */
    const data = location.state.gitHubUserArray;



    useEffect(() => {
        const endOffset = usersOffset + usersPerPage;
        setCurrentUsers(data.slice(usersOffset, endOffset));
        setPageCount((Math.ceil(data.length / usersPerPage)))
    }, [data, usersOffset]);

    const pageClickHandler = (event) => {
        const newOffset = (event.selected * 8) % data.length;
        setUsersOffset(newOffset);
    }

    return (
        <div className={'github-table'}>
            <h1>GitHub User Listing</h1>
            <table className={'gitHubTable'}>
                <thead>
                <tr>
                    <th>Username</th>
                    <th>Repo URL</th>
                    <th>Avatar</th>
                </tr>
                </thead>
                <tbody>
                {data.map((item, index) => {
                    return (
                        <tr key={index}>
                            <td>{item.login}</td>
                            <td>{item.repos_url}</td>
                            <td>{item.avatar_url}</td>
                        </tr>
                    )
                })}
                </tbody>
            </table>
        </div>
    )

}