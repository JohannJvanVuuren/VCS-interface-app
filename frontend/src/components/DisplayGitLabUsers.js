/* Import of the main stylesheet generated by the SCSS files and the preprocessor */
import '../sass/main.css';

/* Import of the useLocation hook which will enable to get the state passed via the Link components in
* the Header component */
import { useLocation } from 'react-router-dom';

/* Definition of the DisplayGitLabUsers component */
export const DisplayGitLabUsers = () => {

    /* Creation of an instance of useLocation */
    const location = useLocation();
    /* Storing the state data from the useLocation instance */
    const data = location.state.gitLabUserArray;

    console.log(data);

    return(
        <div>
            <h2 className="mb-4">
                React Send / Get State using Link Component Example
            </h2>
            <p>Login: {data[0].username}</p>
        </div>
    )
}