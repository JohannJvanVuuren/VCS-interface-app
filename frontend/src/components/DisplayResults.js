import '../sass/main.css';

export const DisplayResults = () => {

}