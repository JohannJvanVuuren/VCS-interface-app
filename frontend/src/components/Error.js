/* Import of main stylesheet generated from SCSS and preprocessor */
import '../sass/main.css';

/* Component to display an error message when the user enters an unknown route */
export const Error = () => {
    return (
        <main className={'error'}>
            <h1>Oops! Page not found!</h1>
        </main>
    )
};
