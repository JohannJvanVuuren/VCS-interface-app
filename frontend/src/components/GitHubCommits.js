/* Import of main stylesheet generated by the SCSS files and preprocessor*/
import '../sass/main.css';
/* Import of axios client to make API call to backend */
import axios from "axios";
/* Import of React Hooks used in this component */
import { useEffect, useState } from "react";
import { useLocation } from 'react-router-dom';

/* Definition of the GitHubCommits component that displays all commits for a specific repo. The commits are in
* descending order so the latest commit will appear at the top of the list */
export const GitHubCommits = () => {

    /* Declaration and initialisation of the state variable needed in this component */
    const [commitsList, setCommitsList] = useState([]);

    /* Use of the useLocation hook to get the state attribute data from the Link component to this page */
    const location = useLocation();
    const user = location.state.user;
    const repoName = location.state.repoName;

    /* The useEffect hook to make the axios call to the backend only when the user or repoName changes */
    useEffect(() => {

        /* The axios call to the githubCommits route on the backend. The username and repo name is encoded in
        * the body of the post request */
        axios.post('http://localhost:8000/githubCommits', {
            user: user,
            repoName: repoName
        })
            .then(response => {
                setCommitsList(response.data)
            })
            .catch(error => {
                console.log('Error', error.message);
            })

    }, [user, repoName]); // useEffect hook's dependencies

    return (
        <div className={'commit-container'}>
            {/* Using the Array.map method to loop through the list of commits received back from the backend
             and render the required properties */}
            {commitsList.map((commit, index) => {
                return (
                    <div key={index}>
                        <p>Date: {commit.commit.author.date} <span>Comment: {commit.commit.message}</span></p>
                    </div>
                )
            })}
        </div>
    )
}