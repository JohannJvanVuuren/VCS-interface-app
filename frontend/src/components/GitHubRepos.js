/* Import of the main stylesheet generated by the SCSS files and preprocessor */
import '../sass/main.css';
import Spinner from 'react-bootstrap/Spinner';

/* Import of useLocation hook to get the state information from the Link component to this component */
import {useLocation, Link} from 'react-router-dom';
import {useEffect, useState} from "react";
import axios from 'axios';

export const GitHubRepos = () => {

    const [repos, setRepos] = useState([]);
    const [isLoading, setIsLoading] = useState(false);

    const location = useLocation();
    const user = location.state.user.login;

    useEffect(() => {

        setIsLoading(true);

        axios.post('http://localhost:8000/githubRepos', {
            user: user
        })
            .then(response => {
                setRepos(response.data);
                setIsLoading(false);
                return response.data;
            })
            .catch(error => {
                console.log(error);
                setIsLoading(false);
            })


    }, [user]);

    return (
        <div>
            <h1>{user}</h1>
            <h2>Repositories</h2>
            {isLoading ?
                <div className={'mx-auto'}>
                    <Spinner animation="border" variant='light'/>
                </div>

             :
                repos.length === 0 ?

                    <h2>No records found</h2>

                :
                repos.map((repo, index) => {
                    return (
                        <div className={'repo-wrapper'} key={index}>
                            <p>{repo.name}</p>
                            <Link
                                to={'/gitHubCommits'}
                                state={{
                                    repoName: repo.name,
                                    user: user
                                }}
                            >
                                Commits
                            </Link>
                        </div>
                    )
                })}

        </div>
    )
}