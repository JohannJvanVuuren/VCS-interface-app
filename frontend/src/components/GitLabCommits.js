import '../sass/main.css';
import { useEffect, useState } from "react";
import { useLocation } from 'react-router-dom';
import axios from "axios";

export const GitHubCommits = () => {

    const [commitsList, setCommitsList] = useState([]);

    const location = useLocation();
    const repoId = location.state.repoId;

    useEffect(() => {

        axios.post('http://localhost:8000/githubCommits', {
            repoId: repoId
        })
            .then(response => {
                setCommitsList(response.data)
            })
            .catch(error => {
                console.log(error);
            })

    }, [repoId]);

    return (
        <div>
            {commitsList.map((commit, index) => {
                return (
                    <div>
                        <p>Date: {commit.comitted_date} <span>Comment: {commit.title}</span></p>
                    </div>
                )
            })}
        </div>
    )
}