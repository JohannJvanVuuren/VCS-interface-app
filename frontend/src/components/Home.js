/* Import of main stylesheet generated by SCSS files and preprocessor */
import '../sass/main.css';

/* The home component. This is the page that will greet the user at the base url */
export const Home = () => {
    return (
        <main className={'home-container'}>
            <section className={'home-content'}>
                <h2 className={'home-heading'}>Version Control System Interface Application</h2>
                <p className={'home-paragraph1'}>
                    This application interfaces with GitHub and GitLab to enable you to search for repositories.
                </p>
                <p className={'home-paragraph2'}>
                    You will be given user and repository details for the selected repository.
                </p>
                <p className={'home-paragraph3'}>
                    We hope you have fun exploring the two VCS providers with this application
                </p>
            </section>
        </main>
    )
}