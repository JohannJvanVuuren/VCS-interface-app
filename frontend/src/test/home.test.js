/* Import of the renderer module */
import renderer from 'react-test-renderer';

/* Import of the component being tested */
import { Home } from '../components/Home';

/* Snapshot test of the Home component */
it('Home component renders correctly', () => {
    const tree = renderer.create(<Home/>).toJSON();
    expect(tree).toMatchSnapshot();
})