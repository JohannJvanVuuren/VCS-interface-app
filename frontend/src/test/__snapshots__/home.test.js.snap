// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<main
  className="home-container"
>
  <section
    className="home-content"
  >
    <h2
      className="home-heading"
    >
      Version Control System Interface: GitHub and GitLab
    </h2>
    <p
      className="home-paragraph1"
    >
      This application interfaces with GitHub and GitLab to enable you to search for repositories.
    </p>
    <p
      className="home-paragraph2"
    >
      You will be given user, repository and commit details for the selected repository.
      <br />
    </p>
    <p
      className="home-paragraph2"
    >
      The latest commit will appear at the top of the list.
    </p>
    <p
      className="home-paragraph3"
    >
      Simply enter the username of the person you want to search for, hit the enter key and
    </p>
    <p
      className="home-paragraph3"
    >
      click on the version control platform (button or icon) that you want to search in.
    </p>
  </section>
</main>
`;
